### PR TITLE
design document for execute hook

### PR DESCRIPTION
**Pull request by nopcoder**
addressing https://github.com/treeverse/lakeFS/issues/2231

### CODE REVIEW

File: design/hook-execution.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Hook Execution
```
Comment (itaiad200) [inline design/hook-execution.md:1]: This entire document assumes we have a list of different applications for hooks that users requested for,  and we prefer to let them do that themselves with webhooks or local execution. Is that the case (i.e. do we have numerous use-cases, or are we're trying to generalise from the start) ?

**Review by itaiad200 — CHANGES_REQUESTED** [review]
I don't think this is needed :)

File: design/hook-execution.md (hunk 1 above)
Comment (nopcoder) [reply design/hook-execution.md:1]: As I understand the use case is the entry level work needed to use the hooks for integration and by enabling a simple execution mechanism will reduce this entry barrier.

**Comment by nopcoder** [comment]
> I don't think this is needed :)

the document for the discussion or the issue?

File: design/hook-execution.md (hunk 1 above)
Comment (itaiad200) [reply design/hook-execution.md:1]: Not sure I understand. This more of a product question. I know of a single user which consider webhooks as a struggle. I also think we can manage without running custom code. Are there any other feedbacks supporting that?

Comment (nopcoder) [reply design/hook-execution.md:1]: Quoted the issue by product.
Will take the discussion with product to sharpen the requirements in order to have better understanding the use-case.

**Comment by nopcoder** [comment]
The issue was updated, closing this one as it is no longer relevant.

**Pull request closed without merging**